### ISSUE TITLE
Namespace end-comment issue with nested namespaces and macros

### ISSUE THREAD

**Issue by CJCombrink** [uncrustify/uncrustify#1565]
Using the latest version of uncrustify, if I have nested namespaces and a multi-line macro definition that contains a closing brace, uncrustify incorrectly adds the closing namespace comment inside the macro instead of at the end of the namespace. 

Using the following option:
```
mod_add_long_namespace_closebrace_comment = 1        # unsigned number
```
With the following file:
```
namespace Namespace1 {
namespace Namespace2 {

  #define SOME_MACRO(  ) \
    if(something) {      \
    } \
    int tmp;

}
}
```

Produces the following wrong output:
```
namespace Namespace1 {
namespace Namespace2 {

  #define SOME_MACRO(  ) \
	if(something) {      \
	} // namespace Namespace2 \
	int tmp;

}
} // namespace Namespace1
```

The expected output is:
```
namespace Namespace1 {
namespace Namespace2 {

  #define SOME_MACRO(  ) \
	if(something) {      \
	} \
	int tmp;

} // namespace Namespace2
} // namespace Namespace1
```

Using version: Uncrustify-0.66.1

I picked this up with version Uncrustify-0.65 but confirmed that this is still an issue in 0.66.1.

Link to debug file: [debug.txt](https://gist.github.com/CJCombrink/378a73b58130d650e826912a8b5003e2)
Generated with: `uncrustify.exe -p debug.txt -c defaults.cfg -f TestMacro.cpp`

[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastbin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"

**Comment by CJCombrink** [comment]
Added file for Uncrustify issue reported here: https://github.com/uncrustify/uncrustify/issues/1565

**Comment by CDanU** [comment]
Thanks for the bug report, the referenced PR fixes your example case.
If you have similar code somewhere else in your codebase it would be nice if you could check if that applies to it as well.

**Comment by guy-maurel** [comment]
The current git version works pretty fine.
  https://github.com/uncrustify/uncrustify
Please test it and report!

**Comment by CJCombrink** [comment]
@gmaurel 
This issue is fixed in 0.67
